docs/workflow/debugging/coreclr/debugging-runtime.md

```diff
@@ -0,0 +1,223 @@
+
+If for some reason `System.Private.CoreLib.dll` is missing, you can rebuild it with the following command, instead of having to go through the whole build again:
+
+```bash
+./build.sh -s clr.corelib+clr.corelibnative -c Debug
+```
+
+**NOTE**: When debugging with _CORE\_LIBRARIES_, the `libs` subset must also be built prior to attempting any debugging.
+
+### SOS on Unix
+
+For Linux and MacOS, you have to install SOS by yourself, as opposed to Windows' Windbg. The instructions are very similar however, and you can find them on these two links:
+
+* The official [Microsoft docs on SOS](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos).
+* The instructions at the [diagnostics repo](https://github.com/dotnet/diagnostics/blob/master/documentation/installing-sos-instructions.md).
+
+It might also be the case that you would need the latest changes in SOS, or you're working with a not-officially-supported scenario that actually works. The most common occurrence of this scenario is when using MacOS Arm64. In this case, you have to build SOS from the diagnostics repo (linked above). Once you have it done, then simply load it to your `lldb`. More details in the following section.
+
+### Debugging CoreCLR with lldb
+
+**NOTE**: Only `lldb` is supported to use with SOS. You can also use `gdb`, `cgdb`, or other debuggers, but you might not have access to SOS.
+
+1. Perform a build of the _clr_ subset of the runtime repo.
+2. Start lldb passing `corerun`, the app to run (e.g. `HelloWorld.dll`), and any arguments this app might need: `lldb /path/to/corerun /path/to/app.dll <app args go here>`
+3. If you're using the installed version of SOS, you can skip this step. If you built SOS manually, you have to load it before starting the debugging session: `plugin load /path/to/built/sos/libsosplugin.so`. Note that `.so` is for Linux, and `.dylib` is for MacOS. You can find more information in the diagnostics repo [private sos build doc](https://github.com/dotnet/diagnostics/blob/main/documentation/using-sos-private-build.md).
+4. Launch program: `process launch -s`
+5. To stop breaks on _SIGUSR1/SIGUSR2_ signals used by the runtime run the following command: `process handle -s false SIGUSR1 SIGUSR2`
+6. Set a breakpoint where CoreCLR is initialized, as it's the most stable point to begin debugging: `breakpoint set -n LoadLibraryExW`.
+7. Get to that point by issuing `process continue` after setting the breakpoint.
+8. Now, you're ready to begin your debugging session. You can set breakpoints or run SOS commands like `clrstack` or `sos VerifyHeap`.  Note that SOS command names are case sensitive.
+
+**NOTE:** _Corerun_ is a simple host that does not support resolving NuGet dependencies. It relies on libraries being locatable via the `CORE_LIBRARIES` environment variable, or present in the same directory as the _corerun_ executable. The instructions above are equally applicable to the _dotnet_ host.
```

Comment:
I would remove most references to `CORE_LIBRARIES`. That is a very uncommon scenario for people. The `corerun` application also has a more discoverable command flag that I would recommend in practice. Either way, neither really need to be mentioned outside of the doc for using `corerun` - see https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/using-corerun.md.